### PR TITLE
feat: Add AWS Workload Identity Federation Support 

### PR DESCRIPTION
**Pull request by ScruffyProdigy**

### CODE REVIEW

File: lib/googleauth/external_account.rb [hunk 1]
```diff
@@ -22,45 +23,91 @@ module Google
...
+      # Provides an entrypoint for all Exernal Account credential classes.
+      class Credentials
+        # The subject token type used for AWS external_account credentials.
+        attr_reader :project_id
+        attr_reader :quota_project_id
+
+        AWS_SUBJECT_TOKEN_TYPE = "urn:ietf:params:aws:token-type:aws4_request".freeze
+        AWS_SUBJECT_TOKEN_INVALID = "aws is the only currently supported external account type".freeze
+
+        TOKEN_URL_PATTERNS = [
```
Comment (lsirac) [inline lib/googleauth/external_account.rb:36]: We actually need to remove the validation for URLs in the config

File: spec/googleauth/oauth2/sts_client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# Copyright 2015 Google, Inc.
```
Comment (lsirac) [inline spec/googleauth/oauth2/sts_client_spec.rb:1]: 2022

File: lib/googleauth/oauth2/sts_client.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# Copyright 2020 Google LLC
```
Comment (lsirac) [inline lib/googleauth/oauth2/sts_client.rb:1]: 2022

File: lib/googleauth/external_account.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
+# Copyright 2015 Google, Inc.
```
Comment (lsirac) [inline lib/googleauth/external_account.rb:1]: 2022

**Review by lsirac — COMMENTED** [review]
Just did a very quick pass. Let's add some documentation for the README

File: lib/googleauth/external_account.rb (hunk 4 above)
Comment (ScruffyProdigy) [reply lib/googleauth/external_account.rb]: Done

File: lib/googleauth/oauth2/sts_client.rb (hunk 3 above)
Comment (ScruffyProdigy) [reply lib/googleauth/oauth2/sts_client.rb]: Done

**Pull request closed without merging**